tests/mbt/model/ccv_sync.qnt

```diff
@@ -0,0 +1,67 @@
+module ccv_sync {
+    import ccv_model.* from "ccv_model"
+    import ccv_types as ccvt from "ccv"
+    import ccv from "ccv"
+    import Time.* from "./libraries/Time"
+    import extraSpells.* from "./libraries/extraSpells"
+
+
+    // The sync module has its own init and step functions.
+    // They ensure that chains do not drift from each other in terms of time,
+    // and all chains produce blocks synchronously.
+    // To do so, it makes use of
+    // "action scheduling", essentially using a variable to
+    // determine the next action to be taken.
+
+    // QueuedEndBlocks contains a list of chains which will end their blocks next,
+    // together with the time advancement they should advance by.
+    // When stepHappy selects an action, it checks if there are any chains in this list,
+    // and if so, it will only select actions that end blocks on the head of the list.
+    // QueuedEndBlocks is thus used to schedule actions.
+    var QueuedEndBlocks: List[(ccvt::Chain, Time)]
+
+    // runs init, then ends and begins a block for each chain, while also starting all consumers.
+    // necessary because we want the happy path to have all consumer chains started.
+    action initHappy =
+        all {
+            init.then(
+                EndAndBeginBlockForProvider(1 * Second, consumerChains, Set())
+            ),
+            QueuedEndBlocks' = consumerChainList.foldl(
+                List(),
+                (acc, consumer) => acc.append((consumer, 1 * Second))
+            ),
+        }
+
+
+
+    // step will advance time for all chains at the same rate,
+    // thus the clock times are always in sync.
+    // This is useful to test happy paths.
+    action stepHappy = any {
```

Comment:
nit: I wonder if a name `stepInSync` or something in this vein would be slightly better because the steps lead to happy paths but the steps themselves are not happy.

Reply:
😢 <- the steps

Jokes aside, good feedback! The Happy name was from an older version, but now with 2 different modules doing different levels of happy things, the change is probably warranted